src/main/java/com/thalesgroup/hudson/plugins/cppcheck/CppcheckSource.java

```diff
@@ -92,18 +92,7 @@ private void buildFileContent() {
             if (tempFile.exists()) {
                 is = new FileInputStream(tempFile);
             } else {
-                // Reading real workspace file is more incorrect than correct,
-                // but the code is left here for backward compatibility with
-                // plugin version 1.14 and less
-                if (cppcheckWorkspaceFile.getFileName() == null) {
-                    throw new IOException("The file doesn't exist.");
-                }
-
-                File file = new File(cppcheckWorkspaceFile.getFileName());
-                if (!file.exists()) {
-                    throw new IOException("Can't access the file: " + file.toURI());
-                }
-                is = new FileInputStream(file);
+                throw new IOException("File doesn't exist: " + tempFile.getAbsoluteFile());
```

Comment:
I removed all backward-compatibility stuff as it is doing more harm than good by giving misleading error message instead of the one which we really care about.